### PR TITLE
Processing model

### PR DESCRIPTION
**Pull request by Lul-Abdifan**
### Blogify
In this milestone,I have done the following

- Create custom methods
  -  Users
      -  A method that returns the 3 most recent posts for a given user.
  -  Posts
      -  A method that updates the posts counter for a user.
      -  A method which returns the 5 most recent comments for a given post.
 -  Comments
     -  A method that updates the comments counter for a post.
-  Likes
     - A method that updates the likes counter for a post

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class User < ApplicationRecord
-  has_many :posts
+  has_many :posts, foreign_key: 'author_id'
   has_many :comments
   has_many :likes
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
 end
```
Comment (zieeco) [inline app/models/user.rb:9]: - Hey, please, kindly check your migration files, there's one which is throwing this error below: kindly fix this issue in your database:


image

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Lul-Abdifan,



Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice processing the data in models ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._




## Optional suggestions

- N/A


**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 1 above)
Comment (Lul-Abdifan) [reply app/models/user.rb:9]: @zieeco I am not getting your feedback. Could you please specify the problem you got when reviewing?

Comment (zieeco) [reply app/models/user.rb:9]: @Lul-Abdifan, I can't run the migration on your database, that's the error it throws. check  [this migration file](https://github.com/Lul-Abdifan/Blog-App/blob/88fc7cc57ae6a68bc8295871ec57f3a56c12318f/db/migrate/20230720133220_rename_old_table_name_to_new_table_name.rb#L1-L5) to fix it

Comment (Lul-Abdifan) [reply app/models/user.rb:9]: @zieeco  I can run from my machine. I rechecked many times and not getting any errors. Could you please check it again?

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Lul-Abdifan , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lul-Abdifan** [comment]
@VICTOR-CHUKWU Thank you

**Pull request merged**